src/cdk/dialog/dialog-ref.ts

```diff
@@ -72,6 +72,13 @@ export class DialogRef<R = unknown, C = unknown> {
         this.close(undefined, {focusOrigin: 'mouse'});
       }
     });
+
+    overlayRef.detachments().subscribe(() => {
+      // Check specifically for `false`, because we want `undefined` to be treated like `true`.
+      if (config.closeOnOverlayDetachments !== false) {
```

Comment:
Optional Nit: Should we just not subscribe if we don't want to close on detachment?

Reply:
This is a pattern we follow in the Material dialog as well. It's in case the config changes while the dialog is open.

Reply:
Ah, I was thinking of that, but saw that the other config field was accessed once. See line 57